pkg/ottl/context_inferrer.go

```diff
@@ -81,11 +83,38 @@ func withContextInferrerPriorities(priorities []string) priorityContextInferrerO
 	}
 }
 
-func (s *priorityContextInferrer) infer(statements []string) (inferredContext string, err error) {
-	s.telemetrySettings.Logger.Debug("Inferring context from statements",
+func (s *priorityContextInferrer) inferFromConditions(conditions []string) (inferredContext string, err error) {
+	return s.innerInfer(conditions, func(condition string) ([]path, map[string]struct{}, map[enumSymbol]struct{}, error) {
+		parsed, err := parseCondition(condition)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		conditionPaths, conditionFunctions, conditionEnums := s.getParsedConditionHints(parsed)
+		return conditionPaths, conditionFunctions, conditionEnums, nil
+	})
+}
+
+func (s *priorityContextInferrer) inferFromStatements(statements []string) (inferredContext string, err error) {
+	return s.innerInfer(statements, func(statement string) ([]path, map[string]struct{}, map[enumSymbol]struct{}, error) {
+		parsed, err := parseStatement(statement)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		statementPaths, statementFunctions, statementEnums := s.getParsedStatementHints(parsed)
+		return statementPaths, statementFunctions, statementEnums, nil
+	})
+}
+
+// hinterFunc is used by the innerInfer function to generate the hints (paths, functions, enums, etc.) for the given OTTL.
+type hinterFunc func(string) ([]path, map[string]struct{}, map[enumSymbol]struct{}, error)
+
+func (s *priorityContextInferrer) innerInfer(ottls []string, hinter hinterFunc) (inferredContext string, err error) {
```

Comment:
nit: given it's not an exported function, I would keep the original name `infer`.